crates/fluvio/src/offset.rs

```diff
@@ -184,9 +184,9 @@ impl Offset {
     /// // Creates an offset pointing 4 places after the oldest log entry
     /// let offset: Offset = Offset::from_beginning(4);
     /// ```
-    pub fn from_beginning(offset: u32) -> Offset {
+    pub fn from_beginning(offset: i64) -> Offset {
```

Comment:
Since it is beginning, this should be positive always.  We have decided to make u32 for now until we decided to this can be changed into make more general:
```
pub fn from_beginning(offset: Impl Into<u32>) -> Offset
```

Reply:
Yes, this was one of the sources of question about the model for offsets. Can leave as `u32` for now. I will create separate issue as mentioned above with my questions about it.